src/lndk_offers.rs

```diff
@@ -135,58 +136,62 @@ pub async fn connect_to_peer(
     Ok(())
 }
 
-#[allow(dead_code)]
-// create_invoice_request builds and signs an invoice request, the first step in the BOLT 12 process of paying an offer.
-pub(crate) async fn create_invoice_request(
-    mut signer: impl MessageSigner + std::marker::Send + 'static,
-    offer: Offer,
-    metadata: Vec<u8>,
-    network: Network,
-    msats: u64,
-) -> Result<InvoiceRequest, OfferError<bitcoin::secp256k1::Error>> {
-    // We use KeyFamily KeyFamilyNodeKey (6) to derive a key to represent our node id. See:
-    // https://github.com/lightningnetwork/lnd/blob/a3f8011ed695f6204ec6a13ad5c2a67ac542b109/keychain/derivation.go#L103
-    let key_loc = KeyLocator {
-        key_family: 6,
-        key_index: 1,
-    };
+impl OfferHandler {
```

Comment:
I know it's tricky across multiple PRs, but I think that this should be in the preceding one so that we don't have to touch it twice, we already do some refactoring for `create_invoice_request`